handlers/jwks/handler.go

```diff
@@ -0,0 +1,29 @@
+package jwks
+
+import (
+	"encoding/json"
+	"github.com/zalando/planb-tokeninfo/keyloader"
+	"log"
+	"net/http"
+)
+
+type jwksHandler struct {
+	loader keyloader.KeyLoader
+}
+
+// Handler creates an http.Handler that provides JWKS responses from the KeyLoader kl
+func Handler(kl keyloader.KeyLoader) http.Handler {
+	return &jwksHandler{loader: kl}
+}
+
+// ServeHTTP serializes the current snapshot of Keys from the KeyLoader as a JSON Web Key Set
+func (h *jwksHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	wrapper := &jwksWrapper{keys: h.loader.Keys()}
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(wrapper); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Println("Failed to finish JWKS response: ", err)
+	} else {
+		w.WriteHeader(http.StatusOK)
```

Comment:
not, sure, i need to look up the what the Encoder.Encode does, but i would bet that it is already too late here to set the status